utility/fft/kissfft_futs.h

```
@@ -0,0 +1,161 @@
+
+#include "kissfft.hh"
+
+template < typename ValueType,
+           int ScaleVal >
+struct kissfft_r2r_fut_t {
+
+    kissfft_r2r_fut_t< ValueType, ScaleVal > (int N) : _N(N) {
+        bool is_inverse_fft = true;
+        fft = new kissfft<ValueType>  (_N*4, !is_inverse_fft);
+        ifft = new kissfft<ValueType> (_N*4, is_inverse_fft);
+
+        fft_in = new std::complex<ValueType> [_N*4];
+        fft_out = new std::complex<ValueType> [_N*4];
+    
+    }
+    
+    virtual ~kissfft_r2r_fut_t < ValueType, ScaleVal > () {
+        free(fft);
+        free(ifft);
+
+        free(fft_in);
+        free(fft_out);
+    }
+
+    template <typename Dimension>
+    void apply(El::Matrix<ValueType>& A, Dimension dimension) const {
+        return apply_impl (A, dimension);
+    }
+
+    template <typename Dimension>
+    void apply_inverse(El::Matrix<ValueType>& A, Dimension dimension) const {
+        // TODO(Jordi): Should we throw an exception?
```

Yes we should throw an exception.